### DISCUSSION TITLE
Accessing the values of <input> elements defined in HTML

### DISCUSSION THREAD

**Discussion by khinsen** [rabbibotton/clog#365] [Q&A] [answered]
Is there a way to access the values of form elements of those form elements have been created from HTML code rather than via clog-create-form-element?

I tried naively:
```
(clog:value (clog:attach-as-child clog-obj html-id))
```
but this fails because `clog:attach-as-child` returns a `clog-element` rather than a `clog-form-element`.

**Comment by rabbibotton** [comment, marked as answer]
(clog:value (clog:attach-as-child clog-obj html-id :clog-type 'clog-form-element))

**Reply by khinsen** [reply]
Simple and effective, thanks!

**Comment by aykaramba** [comment]
Oh wow, okay.  I am taking note of this, thanks for the question and the answer kind sirs.

**Comment by rabbibotton** [comment]
btw - I would use (clog:text-value x) - this is a generic that pics the best way to get the text based on the element - I always mess up with text, value etc and plug-ins use as well for whatever their method is
